File: Assets/Scripts/Player.js
//Basic, barebones class for handling Player movement; this definitely isn't what you payed $50 for

var go : GameObject;
var tform: Transform;
var rb : Rigidbody;

var speed = 24.0;

private var hitCount = 0;

function Awake()
{
	go = gameObject;
	tform = transform;
	rb = GetComponent.<Rigidbody>();
}

function Update () 
{
}

function FixedUpdate () 
{
	var targetVelocity = Vector3(0,Input.GetAxis("Vertical"),  Input.GetAxis("Horizontal"));
	targetVelocity = tform .TransformDirection(targetVelocity);
	targetVelocity *= speed ;
   
	var velocityChange = (targetVelocity - rb.velocity);
	
	rb.AddForce(velocityChange, ForceMode.VelocityChange);
}

function OnTriggerEnter(other : Collider)
{
	hitCount++;
	if(other.CompareTag("Bullet"))
	{	
		Debug.Log("Hit " + hitCount + " times!");
	}
}

File: Assets/Scripts/Player.cs
using UnityEngine;

public class Player : MonoBehaviour
{

    public GameObject go;
    public Transform tform;
    public Rigidbody2D rb;
    public float speed = 24f;

    int hitCount = 0;

    void Awake()
    {
        rb = GetComponent<Rigidbody2D>();
        go = gameObject;
        tform = transform;
    }

    void FixedUpdate()
    {
        var targetVelocity = new Vector3(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"),0);
        targetVelocity = transform.TransformDirection(targetVelocity);
        targetVelocity *= speed;

        var velocityChange = (targetVelocity - (Vector3)rb.velocity);

        rb.AddForce(velocityChange, ForceMode2D.Force);
    }

    void OnTriggerEnter2D(Collider2D other)
    {
        if (other.CompareTag("Bullet"))
        {
            hitCount++;
            Debug.Log("Hit " + hitCount + " times!");
        }
    }

}
